mempool/src/core_mempool/ttl_cache.rs

```diff
@@ -74,7 +70,7 @@ where
         }
     }
 
-    pub fn gc(&mut self, gc_time: SystemTime) {
+    pub fn gc(&mut self, gc_time: Duration) {
```

Comment:
I don't fully understand, why do we change this to duration?

Reply:
This is so `mempool.gc()` can provide a consistent time across `transaction_store` and `metrics_cache` -- `transaction_store` uses Duration. Having these times slightly off can cause some rare races where metrics wouldn't be logged. But more importantly it's harder to reason about.

But, I think I might remove `metrics_cache` in a future PR. Similar to account sequence number, it should really just live with the lifetime of the transaction in transaction store.

Reply:
I meant, gc based on duration for a ttl_cache is a different behavior right? A new entry with shorter duration will be gc earlier than an old entry with longer duration?

Reply:
Previously the lifetime of the items was part of the ttl_cache (`default_timeout`) but now it is a part of the insert function. So yes, this is now a more general expiring cache. However, in it's current usage it still uses a single expiration (`aptos_infallible::duration_since_epoch() + self.system_transaction_timeout`).

High level, this makes it possible to synchronize the caches / items (because they are passed in the expiration time, instead of calling `::now()` independently from within the insert functions.